credentials/credentials_provider.go

```diff
@@ -138,5 +147,21 @@ func (cc *CredentialClient) AcquireCredentials(ctx context.Context, key string,
 
 	span.SetAttributes(attribute.Int64("http.response.size", int64(len(body))))
 
-	return string(body), nil
+	var payload Payload
+	err = json.Unmarshal(body, &payload)
```

Comment:
```suggestion
	err = json.NewDecoder(resp.Body).Decode(&payload)
```

And delete the above `io.ReadAll` lines

Reply:
Changed in [26d4ee3](https://github.com/hasura/ndc-sdk-go/pull/185/commits/26d4ee31bb12d03db81350f6536eea992ec4fd13), but we lose out on the `result.size` span attribute